tls/s2n_connection.c

```diff
@@ -201,8 +205,11 @@ int s2n_connection_wipe(struct s2n_connection *conn)
 
     /* Clone the stuffers */
     /* ignore gcc 4.7 address warnings because dest is allocated on the stack */
+    /* pragma gcc diagnostic was added in gcc 4.6 */
+#if GCC_VERSION >= 40600
```

Comment:
I think we could keep it on one line like "#if defined(__GNUC__) && GCC_VERSION >= 40600". Everything I've read points to the C preprocessor doing short circuit evaluation.